lib/build.ts

```diff
@@ -119,14 +124,14 @@ export async function runBuildTask(
 	// api but it was broken until fixed in
 	// https://github.com/moby/moby/commit/7f334d3acfd7bfde900e16e393662587b9ff74a1
 	// which is why we check for 1.38 here
+	// The image must also have a v2 manifest so that docker won't try to default to x86.
 	const usePlatformOption: boolean =
 		!!task.dockerPlatform &&
-		// workaround for `balena build` with Docker v20.10
-		task.dockerPlatform !== 'none' &&
 		semver.satisfies(
```

Comment:
We now make the calculation here rather than depending on a passed in 'none' override
If we are on a docker version that supports correct use of `--platform`, check to see if there is a v2 manifest.  If there is, then use `--plaform`.